shared/src/middleware.rs

```diff
@@ -71,20 +67,21 @@ impl LoggingInfo {
     }
 }
 
-pub type ProxyLogger = oneshot::Sender<AppOrProxyId>;
+pub type ProxyLogger = mpsc::Sender<AppOrProxyId>;
 
 pub async fn log(
     ConnectInfo(info): ConnectInfo<SocketAddr>,
-    mut req: Request<Body>,
-    next: Next<Body>,
+    mut req: Request,
+    next: Next,
 ) -> Response {
     let method = req.method().clone();
     let uri = req.uri().clone();
     let ip = get_ip(&req, &info);
 
     let mut info = LoggingInfo::new(method, uri, ip);
     // This channel may or may not receive an AppOrProxyId from verify_with_extended_header
-    let (tx, mut rx) = oneshot::channel();
+    // TODO: Solve this with tracing
```

Comment:
I mean I kinda did it already in #162 but because its not merged and I did not want to make the diff bigger I just made this work with a mpsc channel instead of a oneshot as hyper 1 requires request extensions to be clone now.

Reply:
For #162 you messaged me that you still want to change some things. If it is ready for review, please give me a ping.

Reply:
I think its ready for review. I will have to rebase it after this is merged tho